tests/integration/test_fim/test_synchronization/test_registry_responses_win32.py

```diff
@@ -102,75 +103,37 @@ def find_value_in_event_list(key_path, value_name, event_list):
-    Raises:
-        TimeoutError: If an expected event couldn't be captured.
-        ValueError: If a path or value are not in the sync event.
-    """
-    if key_name is None and value_name is None:
-        pytest.skip('key_name and value_name are None. Skipping')
-
-    check_apply_test(tags_to_apply, get_configuration['tags'])
-
-    # Wait for subkey sync messages only when it's expected
-    if key_name is not None:
-        fim.create_registry(fim.registry_parser[key], os.path.join(subkey, key_name), fim.KEY_WOW64_64KEY)
-        fim.check_time_travel(True, monitor=wazuh_log_monitor)
-        wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
-                                callback=fim.callback_detect_end_scan,
-                                error_message='Did not receive expected "end_scan" event').result()
-
-        events = get_sync_msgs(sync_interval + 15)
-        # Parent key sync event
-        assert find_path_in_event_list(test_regs[0], events) is not None, f"No sync event was found for {test_regs[0]}"
 
-        # Subkey sync event
-        subkey_path = os.path.join(test_regs[0], key_name)
-        assert find_path_in_event_list(subkey_path, events) is not None, f'No sync event was found for {subkey_path}'
+def extra_configuration_after_yield():
+    """Remove the registry key when the test ends"""
+    fim.delete_registry(fim.registry_parser[key], subkey, fim.KEY_WOW64_64KEY)
 
-    if value_name is not None:
-        if key_name is not None:
-            key_path = os.path.join(subkey, key_name)
-        else:
-            key_path = subkey
 
-        value_path = os.path.join(key, key_path, value_name)
-
-        key_handle = fim.RegOpenKeyEx(fim.registry_parser[key], key_path, 0, fim.KEY_ALL_ACCESS | fim.KEY_WOW64_64KEY)
-
-        fim.modify_registry_value(key_handle, value_name, fim.REG_SZ, 'This is a test')
-        fim.check_time_travel(True, monitor=wazuh_log_monitor)
-
-        wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
-                                callback=fim.callback_detect_end_scan,
-                                error_message='Did not receive expected "end_scan" event').result()
+@pytest.fixture(scope='function', params=configurations)
+def remove_key_and_restart(request):
+    """Fixture that removes the test key and restart the agent. The aim of this
+       fixture is to avoid false positives if the manager still has the test  key
+       in it's DB.
+    """
+    fim.delete_registry(fim.registry_parser[key], subkey, fim.KEY_WOW64_64KEY)
+    control_service('stop')
+    truncate_file(fim.LOG_FILE_PATH)
+    file_monitor = FileMonitor(fim.LOG_FILE_PATH)
+    setattr(request.module, 'wazuh_log_monitor', file_monitor)
+    control_service('start')
 
-        events = get_sync_msgs(sync_interval + 15)
+    # wait until the sync is done.
+    wazuh_log_monitor.start(timeout=sync_interval + 15, callback=fim.callback_detect_registry_integrity_clear_event,
```

Comment:
I'm not really fond of this hardcoded `+ 15`, can we replace this with something like the default_timeout?